homeassistant/components/input_boolean/__init__.py

```diff
@@ -81,38 +140,26 @@ def is_on(hass, entity_id):
 
     component.async_register_entity_service(SERVICE_TOGGLE, {}, "async_toggle")
 
-    if entities:
-        await component.async_add_entities(entities)
-
     return True
 
 
-async def _async_process_config(config):
-    """Process config and create list of entities."""
-    entities = []
-
-    for object_id, cfg in config[DOMAIN].items():
-        if not cfg:
-            cfg = {}
-
-        name = cfg.get(CONF_NAME)
-        initial = cfg.get(CONF_INITIAL)
-        icon = cfg.get(CONF_ICON)
-
-        entities.append(InputBoolean(object_id, name, initial, icon))
-
-    return entities
+def _filter_yaml(config):
+    """Process config and create list suitable for collection consumption."""
+    return [{CONF_ID: id_, **(conf or {})} for id_, conf in config[DOMAIN].items()]
 
 
 class InputBoolean(ToggleEntity, RestoreEntity):
     """Representation of a boolean input."""
 
-    def __init__(self, object_id, name, initial, icon):
+    def __init__(self, config: typing.Optional[dict], from_yaml: bool = False):
         """Initialize a boolean input."""
-        self.entity_id = ENTITY_ID_FORMAT.format(object_id)
-        self._name = name
-        self._state = initial
-        self._icon = icon
+        self._unique_id = config[CONF_ID]
```

Comment:
I would just store the config and look it up when the property is fetched.

Reply:
I was contemplating storing just the `config`. Decided that having everything in one place would be more readable vs when it is scattered through the class. But I have no objections storing just the config.